edx_event_bus_kafka/internal/consumer.py

```diff
@@ -102,6 +102,9 @@ def inner_from_dict(event_data_dict, ctx=None):  # pylint: disable=unused-argume
             'value.deserializer': AvroDeserializer(schema_str=signal_deserializer.schema_string(),
                                                    schema_registry_client=schema_registry_client,
                                                    from_dict=inner_from_dict),
+            # turn off auto commit to avoid committing offsets before a message has been fully processed. We will
+            # manually commit after processing
```

Comment:
Add mini-ADR in place with highlights of the decision:
1. Note that auto-commit commits the entire batch of messages retrieved, which increases the chances of a message never being processed if the machine dies before finishing the batch.
2. Note the article: https://newrelic.com/blog/best-practices/kafka-consumer-config-auto-commit-data-loss